contracts/helpers/Harvester.sol

```diff
@@ -0,0 +1,185 @@
+    }
+
+    /*//////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+                                                        HARVEST                                                     
+    //////////////////////////////////////////////////////////////////////////////////////////////////////////////////*/
+
+    /// @notice Invests or divests from the yield asset associated to `collateral`
+    /// @dev This transaction either reduces the exposure to `collateral` in the Transmuter or frees up some collateral
+    /// that can then be used for people looking to burn
+    /// @dev Due to potential transaction fees within the Transmuter, this function doesn't exactly bring `collateral`
+    /// to the target exposure
+    function harvest(address collateral) external {
+        (uint256 stablecoinsFromCollateral, uint256 stablecoinsIssued) = TRANSMUTER.getIssuedByCollateral(collateral);
+        CollatParams memory collatInfo = collateralData[collateral];
+        (uint256 stablecoinsFromVault, ) = TRANSMUTER.getIssuedByCollateral(collatInfo.vault);
+        uint8 increase;
+        uint256 amount;
+        uint256 targetExposureScaled = collatInfo.targetExposure * stablecoinsIssued;
+        if (stablecoinsFromCollateral * 1e9 > targetExposureScaled) {
+            // Need to increase exposure to yield bearing asset
+            increase = 1;
+            amount = stablecoinsFromCollateral - targetExposureScaled / 1e9;
+            uint256 maxValueScaled = collatInfo.maxExposureYieldAsset * stablecoinsIssued;
+            // These checks assume that there are no transaction fees on the stablecoin->collateral conversion and so
+            // it's still possible that exposure goes above the max exposure in some rare cases
+            if (stablecoinsFromVault * 1e9 > maxValueScaled) amount = 0;
+            else if ((stablecoinsFromVault + amount) * 1e9 > maxValueScaled)
+                amount = maxValueScaled / 1e9 - stablecoinsFromVault;
+        } else {
+            // In this case, exposure after the operation might remain slightly below the targetExposure as less
+            // collateral may be obtained by burning stablecoins for the yield asset and unwrapping it
+            amount = targetExposureScaled / 1e9 - stablecoinsFromCollateral;
+            uint256 minValueScaled = collatInfo.minExposureYieldAsset * stablecoinsIssued;
+            if (stablecoinsFromVault * 1e9 < minValueScaled) amount = 0;
+            else if (stablecoinsFromVault * 1e9 < minValueScaled + amount * 1e9)
+                amount = stablecoinsFromVault - minValueScaled / 1e9;
+        }
+        if (amount > 0) {
+            try TRANSMUTER.updateOracle(collatInfo.vault) {} catch {}
+
+            rebalancer.adjustYieldExposure(
+                amount,
+                increase,
+                collateral,
+                collatInfo.vault,
+                (amount * (1e9 - maxSlippage)) / 1e9
```

Comment:
This slippage protection doesn't make sense when it is not the user paying for the swaps

Reply:
It's just so rebalances don't occur when we're in situations with badly calibrated max exposures and prices